### DISCUSSION TITLE
Image styling not working as expected in Next.js 13

### DISCUSSION THREAD

**Discussion by kendjr** [vercel/next.js#44486] [Help] [answered]
I would like to have responsive images after upgrading to Next.js 13, and the Image component will not accept the "position: relative" style. The [Next.js docs][1] instruct you to use the fill boolean and assign the position in the parent, but that is not working. I attempted to assign the style in the parent and in the actual component but neither is working.

Is this possible, or is there a better way to have responsive images?

The below div tag is inside of a component:

       <div className="row">
        {stories.map((story) => {
          return (
            <div key={story.id} className={"col-sm-4 " + styles.imageStyle}>
              <Image
                src={story.image}
                alt={story.alt}
                fill
                className={styles.imageStyle}
              />
            </div>
          );
        })}
      </div>

And styles is defined in a module.scss file

      .imageStyle {
    object-fit: cover;
    position: relative;
    }

inspecting the code shows the Image is overriding the style

    element.style {
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    color: transparent;
    }

    .story-images-block_imageStyle__RCYiz {
    object-fit: cover;
    *position: relative; (*this has a strikethrough)
    }

  [1]: https://nextjs.org/docs/api-reference/next/image#fill

**Comment by Xevion** [comment, marked as answer]
You aren't supposed to assign the `position: relative` attribute to the `Image` component, you are intended to assign it to the parent element.

```jsx
<div className="relative">
    <Image src="/image.jpeg" fill alt="" className="object-cover" />
</div>
```

Tailwind classes are used here. If you could show the JSX/CSS you are using for the parent element, that could help narrow down the solution, but what you've shown applies the `position: relative` styling only to the Image component, which won't work at all.

**Reply by kendjr** [reply]
Isn't the above 
`<div key={story.id} className={"col-sm-4 " + styles.imageStyle}> `
the parent element? I assign position relative there as well as within the Image component.
This is the entire code. The styles.imageStyle assigns both objectfit and position as shown above.

Thanks!

`return ( <div className="container text-center"> <div className="row"> {stories.map((story) => { return ( <div key={story.id} className={"col-sm-4 " + styles.imageStyle}> <Image src={story.image} alt={story.alt} fill /> </div> ); })} </div> </div> );`

**Reply by Xevion** [reply]
What you have appears to be correct, but you haven't actually detailed what is 'not working as expected'. Do you have any screenshots that detail what you expect v.s. what is wrong?

**Reply by kendjr** [reply]
Screenshot one has the parent element highlighted, and you can see position: relative. The images do not appear on the screen

![parent](https://user-images.githubusercontent.com/3506293/210362962-95429792-a2a1-4444-a212-c3946d00fb6b.png)


---------------------
Screenshot two has the img element highlighted, and you can see it overrides the parent with position: absolute.

![imgelement](https://user-images.githubusercontent.com/3506293/210362959-971be445-df26-40f1-aee8-6409b0a02eda.png)


---------------------
Screenshot three I manually override the position style while highlighting the img element using the chrome tool, and the images appear correctly.

![forcing](https://user-images.githubusercontent.com/3506293/210362957-eee7677d-8332-40e5-bb38-f9ba81d11801.png)

Thanks!
Ken

**Reply by Xevion** [reply]
Can you add an `object-fit: cover` or `object-fit: contain` style to the image?

**Reply by kendjr** [reply]
yes, and it will make that part of the image element style.  However, it has no effect because position is still "absolute".

**Reply by Xevion** [reply]
With `object-fit: cover`
![With object-fit-cover](https://user-images.githubusercontent.com/44609630/210441682-40f792e4-8425-4829-9aca-5f32b8bda3e7.png)

Without `object-fit: cover`
![Without object-fit-cover](https://user-images.githubusercontent.com/44609630/210441679-1e0d4e25-cadb-493f-8a4b-d73385ff8605.png)

**Reply by kendjr** [reply]
I just noticed that your image tag has a class, mine does not. Also, when I highlight image, I do not see the .story-images-block at all
![nostyle](https://user-images.githubusercontent.com/3506293/210443972-9b053571-121b-4717-8d4e-30b0723e1477.png)

**Reply by kendjr** [reply]
I rewrote the component without using "fill" and it still completely ignores the style of the parent.  Is there a setting that tells the image to overwrite all parent styles? As you can see above it operates as though it is assigning inline styles to the image component, and completely ignoring any custom styles I have for that page.

**Reply by kendjr** [reply]
The issue is the height of the image - it is always zero. If I insert text inside of the container, on a peer level to the image element, the height of the image changes to the height of the text.
`      <div className={"container " + styles.imageStyle}>
        <Image
          src="/img/stories/b/pepperstreetpets.png"
          fill
          sizes="(max-width: 768px) 100vw,
              (max-width: 1200px) 50vw,
              33vw"
          style={{ objectFit: "cover" }}
          alt="Pepper Street Pets"
        />
        <div className="lh-lg">this is a sentence inside of the image container</div>
      </div>
`
After inserting the div with the sentence above, the image size changed from zero to 32px
Could this have something to do with Bootstrap?

**Reply by kendjr** [reply]
@leerob can you take a look at this? I am stumped.

**Reply by leerob** [reply]
The original answer here seems accurate - could you try pulling this out into a small reproducible example? It's hard to tell what other styles you might have on the page (e.g. global styles). A CodeSandbox link would be great.

**Reply by kendjr** [reply]
Thanks, I appreciate the look. I am not that familiar with CodeSandbox but will try. I have always had "position: relative" assigned to the parent, but the Image component is ignoring it for some reason.

FWIW, this works fine if I simply don't use the Image component and simply use the old standard <img> tag. 

I don't understand how the global styles could interfere, since based on the Chrome Inspector the parent div has the "position: relative" correctly assigned, and it is active, but I will try to pull out a portion of this as you suggested.

Thanks again.

**Reply by kendjr** [reply]
@leerob I put the code in CodeSandbox, and I am getting the same behavior.
https://codesandbox.io/p/sandbox/billowing-monad-el8qpe?file=%2Fpages%2Findex.tsx&selection=%5B%7B%22endColumn%22%3A13%2C%22endLineNumber%22%3A17%2C%22startColumn%22%3A13%2C%22startLineNumber%22%3A17%7D%5D
Please let me know if that works!

**Reply by Xevion** [reply]
![](https://i.xevion.dev/2023/01/firefox_IEgT5zpVKp.png)

[Preview](https://kdlcyk-3000.preview.csb.app/) | [Source](https://codesandbox.io/p/sandbox/fancy-sunset-kdlcyk)

This was a bit finnicky and weird, but I got it working. Seems like the main issue present was that you didn't specify any kind of height for the elements to expand. With a height of 100%, the image will only be as large as it's parent.

**Reply by kendjr** [reply]
Thanks! That is very helpful and appreciated!
If I increase the height to allow a wider image, it has too much space on top and bottom of the image when the screen size decreases, but I believe I should be able to fix with CSS.
I am assuming my design with a single image across the screen is not very popular, which is why more are not experiencing this issue.
